pkg/resources/resource.go

```diff
@@ -27,4 +30,16 @@ type FlavorResource struct {
 	Resource corev1.ResourceName
 }
 
+func (fr FlavorResource) String() string {
+	return fmt.Sprintf(`{"Flavor":"%s","Resource":"%s"}`, string(fr.Flavor), string(fr.Resource))
+}
+
 type FlavorResourceQuantities map[FlavorResource]int64
+
+func (q FlavorResourceQuantities) MarshalJSON() ([]byte, error) {
+	temp := make(map[string]int64)
```

Comment:
```suggestion
	temp := make(map[string]int64, len(q))
```
nit, just to avoid unnecessary resizes

Reply:
done, thanks